src/index.ts

```diff
@@ -1,3 +1,166 @@
 // Copyright (c) Jupyter Development Team.
 // Distributed under the terms of the Modified BSD License.
 'use-strict';
+
+import {
+  IInputAreaViewModel
+} from 'jupyter-js-input-area';
+
+import {
+  IOutputAreaViewModel
+} from 'jupyter-js-output-area';
+
+import {
+  IObservableList
+} from 'phosphor-observablelist';
+
+import {
+  Widget
+} from 'phosphor-widget';
+
+import './index.css';
+
+
+/**
+ * An enum which describes the type of cell.
+ */
+enum CellType {
+  /**
+   * The cell contains code input.
+   */
+  Code,
+
+  /**
+   * The cell contains markdown.
+   */
+  Markdown,
+
+  /**
+   * The cell contains raw text.
+   */
+  Raw
+}
+
+
+/**
+ * The arguments object emitted with the `stateChanged` signal.
+ */
+export
+interface ICellChangedArgs<T> {
+  name: string,
+  oldValue: T;
+  newValue: T;
+}
+
+
+/**
+ * The definition of a model object for a base cell.
+ */
+interface IBaseCellViewModel {
+
+  /**
+   * The type of cell.
+   */
+  cellType: CellType;
```

Comment:
I just grabbed it from the existing code where is it `cell_type`...I think
that `type` is a bit vague...

On Tue, Nov 3, 2015 at 9:12 PM, Matthias Bussonnier <
notifications@github.com> wrote:

> In src/index.ts
> <https://github.com/jupyter/jupyter-js-cells/pull/2#discussion_r43835466>:
>
> > +interface ICellChangedArgs<T> {
> > +  name: string,
> > +  oldValue: T;
> > +  newValue: T;
> > +}
> > +
> > +
> > +/**
> > + * The definition of a model object for a base cell.
> > + */
> > +interface IBaseCellViewModel {
> > +
> > +  /**
> > +   * The type of cell.
> > +   */
> > +  cellType: CellType;
>
> cellType or just type? otherwise it will read cell.cellType which seem a
> bit redundant.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/jupyter-js-cells/pull/2/files#r43835466>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com